### PR TITLE
feat: APP-2730 - Implement InputDate component

### PR DESCRIPTION
**Pull request by RuggeroCino**
## Description

- Implement InputDate component
- Fix disabled input style on Firefox

Task: [APP-2730](https://aragonassociation.atlassian.net/browse/APP-2730)

## Type of change

-   [x] New feature (non-breaking change which adds functionality)

## Checklist:

-   [x] I have selected the correct base branch.
-   [x] I have performed a self-review of my own code.
-   [x] I have commented my code, particularly in hard-to-understand areas.
-   [x] I have made corresponding changes to the documentation.
-   [x] My changes generate no new warnings.
-   [x] Any dependent changes have been merged and published in downstream modules.
-   [x] I ran all tests with success and extended them if necessary.
-   [x] I have updated the `CHANGELOG.md` file in the root folder of the package after the [UPCOMING] title and before
        the latest version.
-   [x] I have tested my code on the test network.

[APP-2730]: https://aragonassociation.atlassian.net/browse/APP-2730?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -223,6 +223,7 @@ module.exports = {
     plugins: [
         require('tailwindcss/plugin')(({ addVariant }) => {
             addVariant('search-cancel', '&::-webkit-search-cancel-button');
+            addVariant('calendar-icon', ['&::-webkit-calendar-picker-indicator', '&::-webkit-inner-spin-button']);
```
Comment (Fabricevladimir) [inline tailwind.config.js:226]: Don't forget the non-chromium browsers like Firefox!

File: src/components/input/inputDate/inputDate.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    const handleCalendarClick = () => {
+        inputRef.current?.showPicker();
+    };
+
+    return (
+        <InputContainer {...containerProps}>
+            <input
+                type="date"
+                className={classNames('calendar-icon:hidden calendar-icon:appearance-none', inputClassName)}
```
Comment (Fabricevladimir) [inline src/components/input/inputDate/inputDate.tsx:26]: For the warning and critical variants, the color of the value should be changed to `warning-800` and `critical-800` respectively. This only seems to apply to the non default value/placeholder of 'mm/dd/yyyy'.

Also, on Firefox both icons are shown (see comment on the TW config file), and when the input is disabled, the `InputContainer` background color needs to be set to the same as the actual date input. Might be that the `InputContainer` component disabled color itself needs to be updated, not sure.

File: src/components/input/inputDate/inputDate.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import classNames from 'classnames';
+import { forwardRef, useRef } from 'react';
+import { mergeRefs } from 'react-merge-refs';
+import { Button } from '../../button';
+import { IconType } from '../../icon';
+import { InputContainer, type IInputComponentProps } from '../inputContainer';
+import { useInputProps } from '../useInputProps';
+
+export interface IInputDateProps extends IInputComponentProps {}
```
Comment (Fabricevladimir) [inline src/components/input/inputDate/inputDate.tsx:9]: `IInputComponentProps` exposes a `maxLength` prop; I'm not sure it makes sense to have that for a date picker. What do you think?

File: src/components/input/inputDate/inputDate.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    return (
+        <InputContainer {...containerProps}>
+            <input
+                type="date"
+                className={classNames('calendar-icon:hidden calendar-icon:appearance-none', inputClassName)}
+                ref={mergeRefs([inputRef, ref])}
+                {...otherInputProps}
+            />
+            <Button
```
Comment (Fabricevladimir) [inline src/components/input/inputDate/inputDate.tsx:30]: Don't forget to pass in the `disabled` prop down to the button.

**Review by Fabricevladimir — COMMENTED** [review]
Thanks! Left a couple of comments

File: src/components/input/inputDate/inputDate.tsx (hunk 4 above)
Comment (RuggeroCino) [reply src/components/input/inputDate/inputDate.tsx:31]: Good catch, adding a test as well 👍🏻

File: src/components/input/inputDate/inputDate.tsx (hunk 3 above)
Comment (RuggeroCino) [reply src/components/input/inputDate/inputDate.tsx]: Agree, I'm removing it

File: tailwind.config.js (hunk 1 above)
Comment (RuggeroCino) [reply tailwind.config.js:227]: Good catch 👌🏻 It turns out you cannot style the calendar icon on Firefox (https://bugzilla.mozilla.org/show_bug.cgi?id=1812397). I updated the implementation to hide the default icon using the custom one

File: src/components/input/inputDate/inputDate.tsx (hunk 2 above)
Comment (RuggeroCino) [reply src/components/input/inputDate/inputDate.tsx]: - I updated the input-container to set the correct color for values depending on the variant ✅
- Fixed the disabled input style on Firefox ✅
- Time and Date input placeholders will currently use the same color as the value, will discuss this with designer ⌛

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,15 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
 ## [Unreleased]
 
+### Added
+
+-   Implement `InputDate` component
+
+### Fixed
+
+-   Fix input value colors on `warning` and `critical` variants
+-   Fix disabled input style on Firefox
```
Comment (Fabricevladimir) [inline CHANGELOG.md:17]: Considering this is the "first" implementation, don't know if it makes sense to include a "Fixed" section for the `InputDate`

**Review by Fabricevladimir — APPROVED** [review]
Looking fly! Thanks

File: CHANGELOG.md (hunk 5 above)
Comment (RuggeroCino) [reply CHANGELOG.md:17]: Actually the fix for the disabled input style on Firefox was on the `useInputProps` component, we did fix the style on the other inputs 👌🏻

**Pull request merged**